shared/naturalcrit/codeEditor/codeEditor.jsx

```diff
@@ -18,6 +18,16 @@ if(typeof navigator !== 'undefined'){
 	//Addons
 	require('codemirror/addon/fold/foldcode.js');
 	require('codemirror/addon/fold/foldgutter.js');
+	require('codemirror/addon/fold/xml-fold.js');
+	require('codemirror/addon/search/search.js');
+	require('codemirror/addon/search/searchcursor.js');
+	require('codemirror/addon/search/jump-to-line.js');
+	require('codemirror/addon/search/match-highlighter.js');
+	require('codemirror/addon/search/matchesonscrollbar.js');
+	require('codemirror/addon/dialog/dialog.js');
```

Comment:
Can you clarify to me what each of these does?

Reply:
```javascript
    require('codemirror/addon/search/search.js');
	require('codemirror/addon/search/searchcursor.js');
	require('codemirror/addon/search/jump-to-line.js');
	require('codemirror/addon/search/match-highlighter.js');
	require('codemirror/addon/search/matchesonscrollbar.js');
	require('codemirror/addon/dialog/dialog.js');
```
All of the above are for find/replace. 
https://codemirror.net/demo/search.html
https://codemirror.net/doc/manual.html#addon_matchesonscrollbar

```javascript
	require('codemirror/addon/fold/xml-fold.js');
	require('codemirror/addon/edit/closetag.js');
```
These are for auto-closing html tags

```javascript
	require('codemirror/addon/edit/trailingspace.js');
```
This adds a css class to trailing spaces, which allows us to customize it to our liking

```javascript
	require('codemirror/addon/selection/active-line.js');
```
This highlights the active line